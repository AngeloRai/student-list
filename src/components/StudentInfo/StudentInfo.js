const StudentInfo = ({ student }) => {
  return (
    <div className="text-sm ">
      <div>City: {student.city}</div>
      <div>Email: {student.email}</div>
      <div>Company: {student.company}</div>
      <div>Skills: {student.skill}</div>
      <div>
        Average:{" "}
        {(
          student.grades.reduce((a, b) => Number(a) + Number(b)) /
          student.grades.length
        ).toFixed(2)}
        %
      </div>
    </div>
  );
};

export default StudentInfo;
