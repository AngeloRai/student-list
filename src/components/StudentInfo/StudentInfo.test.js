import React from "react";
import { shallow, mount } from "enzyme";
import StudentInfo from "./StudentInfo";
import toJson from "enzyme-to-json";

describe("tests student info component", () => {
  const mockStudent = {
    city: "New York",
    email: "student@student.com",
    company: "company",
    skills: "genius",
    grades: [1, 2, 3, 4]
  };
  
  it("renders without crashing", () => {
    const wrapper = shallow(<StudentInfo student={mockStudent}/>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("accepts props", () => {
    const wrapper = mount(<StudentInfo student={mockStudent} />);
    expect(wrapper.props().student.city).toBe(mockStudent.city);
    expect(wrapper.props().student.email).toBe(mockStudent.email);
    expect(wrapper.props().student.company).toBe(mockStudent.company);
    expect(wrapper.props().student.skills).toBe(mockStudent.skills);
    expect(wrapper.props().student.grades.length).toBe(4);
    
  });
});
