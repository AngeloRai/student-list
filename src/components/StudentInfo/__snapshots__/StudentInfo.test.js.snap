// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests student info component renders without crashing 1`] = `
<div
  className="text-sm "
>
  <div>
    City: 
    New York
  </div>
  <div>
    Email: 
    student@student.com
  </div>
  <div>
    Company: 
    company
  </div>
  <div>
    Skills: 
  </div>
  <div>
    Average:
     
    2.50
    %
  </div>
</div>
`;
